databroker/tests/test_broker.py

```diff
@@ -389,6 +389,11 @@ def test_filters(db_empty, RE):
 
     assert header['start']['uid'] == ken_calib_uid
 
+    db.clear_filters()
+    db.add_filter(start_time='2017')
+    db.add_filter(start_time='2017')
```

Comment:
I think we still need to consider the case 
```
db.add_filter(start_time='2017')
db.add_filter(start_time='2017', stop_time='2018). 
```
So basically we will raise error if there are duplicated keys. Users can clean the filters and search again. Also per discussion with @tacaswell